synfig-studio/src/gui/docks/dock_timetrack2.cpp

```diff
@@ -233,6 +233,29 @@ void Dock_Timetrack2::setup_tool_palette()
 		action_button_map[tool_button->get_name()] = tool_button;
 		tool_item_group->add(*tool_button);
 	}
+
+	std::vector<std::string> stock_id{ "synfig-interpolation_type_clamped", "synfig-interpolation_type_tcb",
+								  "synfig-interpolation_type_const", "synfig-interpolation_type_ease" ,
+								  "synfig-interpolation_type_linear"};
+	std::vector<std::string> name{ "Clamped", "TCB" , "Constant" , "Ease In/Out" , "Linear"};
+	std::vector<synfig::Interpolation> interp_type{ synfig::INTERPOLATION_CLAMPED, synfig::INTERPOLATION_TCB,
+												  synfig::INTERPOLATION_CONSTANT, synfig::INTERPOLATION_HALT,
+												  synfig::INTERPOLATION_LINEAR};
+	//button1
+	for (int i=0; i<=4; i++) {
+		Gtk::ToolButton *tool_button = manage(new Gtk::ToolButton(
+														*manage(new Gtk::Image(
+																	Gtk::StockID(stock_id[i]),
+																	Gtk::IconSize::from_name("synfig-small_icon_16x16"))),
+																	_(name[i].c_str())));
+		tool_button->signal_clicked().connect( [this, interp_type, i](){
+
+			current_widget_timetrack->interpolate_selected(interp_type[i]);
+
+		});

```

Comment:
Please avoid lambda signal slot to avoid crash.
Use `sigc::memfun()`. You may use `sigc::bind()` to pass a variable as signal slot (class method) argument.

Reply:
I tried doing so before committing as I remembered what you said earlier about the complications that come with the lambda but for some reason, it was giving out seg errors.

Reply:
hmm ok, I used the "safe" alternative for the current lambda which is using sigc::track_obj with it as mentioned in the [link ](https://gitlab.gnome.org/GNOME/gtkmm-documentation/-/issues/10)you provided earlier, works like a charm :).

Reply:
Maybe, I can't remember. Maybe I was just experimenting `sigc::track_obj` XD